src/run.ts

```diff
@@ -0,0 +1,67 @@
+import * as core from '@actions/core';
+import {spawn} from 'child_process';
+import {parse} from 'shell-quote';
+
+type spawnResult = {
+    code: number | null
+    stdout: string | null
+    stderr: string | null
+}
+
+export class CLIError extends Error {
+    constructor(message: string) {
+        super(message);
+        this.name = "CLIError";
+    }
+}
+
+
+const spawnPromise = async (command: string, args: string[]): Promise<spawnResult> => new Promise((resolve, reject) => {
+    const process = spawn(command, args);
+    process.stdout.setEncoding('utf-8');
+    process.stderr.setEncoding('utf-8');
+
+    process.on('exit', code => {
+        resolve({
+            code,
+            stdout: process.stdout.read() as string | null,
+            stderr: process.stderr.read() as string | null,
+        });
+    });
+    process.on('error', err => {
+        reject(err);
+    });
+});
```

Comment:
```suggestion
const spawnPromise = util.promisify(spawn);
```

(Need to import [util](https://nodejs.org/api/util.html#utilpromisifyoriginal))